### ISSUE TITLE
Updating to 4.1.1 from 4.1.0 via NuGet Causes Errors

### ISSUE THREAD

**Issue by jblacker** [NLog/NLog#918]
I'm not completely sure if this is an isolated incident, but as soon as I updated from 4.1.0 to 4.1.1 via NuGet I received a "Could not load file or assembly 'NLog' or one of its dependencies" error message.  I do have the NLog.Web package installed (this is a ASP.NET MVC application).  I attempted to remove all NLog packages.  Reinstall all of them, only have the main NLog package, and even attempted to revert to the 4.1.0 package without NLog.Web, but alas nothing has seemed to resolve it.  Any idea what may cause this, or is this potentially an issue with how it's currently loaded on NuGet?  Everything was working swimmingly prior to the update.

Here's the full Message:
```
Server Error in '/' Application.

Could not load file or assembly 'NLog' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: System.IO.FileLoadException: Could not load file or assembly 'NLog' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Source Error: 

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Assembly Load Trace: The following information can be helpful to determine why the assembly 'NLog' could not be loaded.


=== Pre-bind state information ===
LOG: DisplayName = NLog
 (Partial)
WRN: Partial binding information was supplied for an assembly:
WRN: Assembly Name: NLog | Domain ID: 17
WRN: A partial bind occurs when only part of the assembly display name is provided.
WRN: This might result in the binder loading an incorrect assembly.
WRN: It is recommended to provide a fully specified textual identity for the assembly,
WRN: that consists of the simple name, version, culture, and public key token.
WRN: See whitepaper http://go.microsoft.com/fwlink/?LinkId=109270 for more information and common solutions to this issue.
LOG: Appbase = file:///C:/Users/jblacker/Development/DigiscribePCS/DigiscribePCS/
LOG: Initial PrivatePath = C:\Users\jblacker\Development\DigiscribePCS\DigiscribePCS\bin
Calling assembly : (Unknown).
===
LOG: This bind starts in default load context.
LOG: Using application configuration file: C:\Users\jblacker\Development\DigiscribePCS\DigiscribePCS\web.config
LOG: Using host configuration file: C:\Users\jblacker\Documents\IISExpress\config\aspnet.config
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework64\v4.0.30319\config\machine.config.
LOG: Policy not being applied to reference at this time (private, custom, partial, or location-based assembly bind).
LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/Temporary ASP.NET Files/vs/590cb163/b09c6c64/NLog.DLL.
LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/Temporary ASP.NET Files/vs/590cb163/b09c6c64/NLog/NLog.DLL.
LOG: Attempting download of new URL file:///C:/Users/jblacker/Development/DigiscribePCS/DigiscribePCS/bin/NLog.DLL.
LOG: Using application configuration file: C:\Users\jblacker\Development\DigiscribePCS\DigiscribePCS\web.config
LOG: Using host configuration file: C:\Users\jblacker\Documents\IISExpress\config\aspnet.config
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework64\v4.0.30319\config\machine.config.
LOG: Redirect found in application configuration file: 4.1.0.0 redirected to 4.1.1.0.
LOG: Post-policy reference: NLog, Version=4.1.1.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c
LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/Temporary ASP.NET Files/vs/590cb163/b09c6c64/NLog.DLL.
LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/Temporary ASP.NET Files/vs/590cb163/b09c6c64/NLog/NLog.DLL.
LOG: Attempting download of new URL file:///C:/Users/jblacker/Development/DigiscribePCS/DigiscribePCS/bin/NLog.DLL.
WRN: Comparing the assembly name resulted in the mismatch: Build Number
ERR: Failed to complete setup of assembly (hr = 0x80131040). Probing terminated.

Stack Trace: 


[FileLoadException: Could not load file or assembly 'NLog' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]

[FileLoadException: Could not load file or assembly 'NLog, Version=4.1.1.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]
   System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +0
   System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +210
   System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean forIntrospection) +242
   System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, Boolean forIntrospection) +17
   System.Reflection.Assembly.Load(String assemblyString) +35
   System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +122

[ConfigurationErrorsException: Could not load file or assembly 'NLog, Version=4.1.1.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]
   System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +12496980
   System.Web.Configuration.CompilationSection.LoadAllAssembliesFromAppDomainBinDirectory() +499
   System.Web.Configuration.AssemblyInfo.get_AssemblyInternal() +131
   System.Web.Compilation.BuildManager.GetReferencedAssemblies(CompilationSection compConfig) +331
   System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath, Boolean& isRefAssemblyLoaded) +148
   System.Web.Compilation.BuildManager.ExecutePreAppStart() +172
   System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException) +1151

[HttpException (0x80004005): Could not load file or assembly 'NLog, Version=4.1.1.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +12618692
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +159
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +12458309
```

**Comment by 304NotModified** [comment]
Yes it's too bad somethings we cannot prevent easily. The problem is  described at http://nlog-project.org/2015/09/12/nlog-4-1-1-has-been-released.html

Tl;dr
Remove  `<bindingRedirect>` from your `.config`

**Comment by jblacker** [comment]
Sorry, I should have done more research before posting.  =(

**Comment by 304NotModified** [comment]
No problem! I would find it also unexpected as an user.

**Comment by 304NotModified** [comment]
If you have more questions, just let us know!
